services/graphql-server/src/graphql/resolvers/email/newsletter.js

```diff
@@ -17,10 +18,26 @@ module.exports = {
         attributes,
       } = input;
 
+      const existingNewsletterProduct = await basedb.findOne('platform.Product', { _id: id }, { projection: { provider: 1 } });
```

Comment:
Just call this `product`, in this context there's no need to distinguish it from other product types, or a newly created one. Also, use the `strictFindById` method here.